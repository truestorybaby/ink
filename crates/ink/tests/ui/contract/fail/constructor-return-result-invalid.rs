#[ink::contract]
mod contract {
    #[ink(storage)]
    pub struct Contract {}

    #[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
    #[cfg_attr(feature = "std", derive(::scale_info::TypeInfo))]
    pub enum Error {
        Foo,
    }

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Result<u8, Error> {
            Ok(5_u8)
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
