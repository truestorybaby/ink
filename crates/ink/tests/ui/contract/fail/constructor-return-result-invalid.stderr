error[E0277]: the trait bound `ConstructorOutputValue<Result<u8, contract::Error>>: ConstructorOutput<Contract>` is not satisfied
  --> tests/ui/contract/fail/constructor-return-result-invalid.rs:14:9
   |
14 |         pub fn constructor() -> Result<u8, Error> {
   |         ^^^ the trait `ConstructorOutput<Contract>` is not implemented for `ConstructorOutputValue<Result<u8, contract::Error>>`
   |
   = help: the following other types implement trait `ConstructorOutput<C>`:
             ConstructorOutputValue<C>
             ConstructorOutputValue<Result<C, E>>
